File: app/models/api/Scorable.java
// SPDX-FileCopyrightText: 2024 The members of the EXAM Consortium
//
// SPDX-License-Identifier: EUPL-1.2

package models.api;

import com.fasterxml.jackson.databind.JsonNode;
import java.util.Optional;
import play.mvc.Result;

public interface Scorable {
    Double getAssessedScore();
    Double getMaxAssessedScore();
    boolean isRejected();
    boolean isApproved();
    Optional<Result> getValidationResult(JsonNode node);
}

File: app/models/sections/Scorable.scala
// SPDX-FileCopyrightText: 2024 The members of the EXAM Consortium
//
// SPDX-License-Identifier: EUPL-1.2

package models.sections

trait Scorable:
  def getAssessedScore: Double
  def getMaxAssessedScore: Double
  def isRejected: Boolean
  def isApproved: Boolean
